### PR TITLE
added feature to transfer pokemon after maxCP ratio

### PR DESCRIPTION
**Pull request by pashorizer**
Added option to transfer pokemon with CpMax ratio
looks like this

`Going to transfer NIDORAN_FEMALE with CP 327 and IV 62%; reason: CP < 1500 and IV < 89% and CP max 704: achieved 46% <= 60%`

One use case would be like this:

```
transfer_cp_threshold=-1
transfer_iv_threshold=89 // whatever you like as IV
transfer_cp_min_threshold=60 
```
If that Nidoran had 600CP it would result in achieved CP% ~80% and would pass the filter. So no need to use CP anymore since with that you would always filter out a Magikarp which can have max ~260CP anyways if you set `transfer_cp_threshold` above 260. Which most of us did.

OR if you want to have the very best
```
transfer_cp_threshold=-1
transfer_iv_threshold=-1
transfer_cp_min_threshold=90
```
With that setting every Pokemon you catch must have almost a perfect IV and almost max possible CP for the current trainer lvl.

### CODE REVIEW

**Comment by yscphdk** [comment]
How do I add it?

**Comment by pashorizer** [comment]
@yscphdk Checkout on my branch

File: src/main/kotlin/ink/abb/pogo/scraper/util/pokemon/PokemonData.kt [hunk 1]
```diff
@@ -22,6 +22,11 @@ fun PokemonData.getIvPercentage(): Int {
     return ivPercentage
 }
 
+fun PokemonData.getCpPercentageToPlayer(playerlvel: Int): Int {
```
Comment (jabbink) [inline src/main/kotlin/ink/abb/pogo/scraper/util/pokemon/PokemonData.kt]: `playerlevel`

**Comment by jabbink** [comment]
Would make sorting on this make sense?

**Comment by andredp** [comment]
@jabbink, I think sorting by max CP already does the same sorting as %maxCP... Because if a pokemon has more %maxCP that another from the same species it will have higher CP, right? Or am I missing something?

**Comment by andredp** [comment]
I was answering to @jabbink 's question. I agree with you that among different pokemons, %maxCP is a way better measurement of how good the pokemon is, but the sorting feature is only used between pokemons of the same species (when you reach the maximum amount of that pokemon, it decides which one will be transferred, in which case looking at the CP alone is enough.)

**Comment by pashorizer** [comment]
Dat update button

**Pull request merged**